src/link-manager.js

```diff
@@ -0,0 +1,198 @@
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+
+/**
+ * Settings for link views.
+ * This part will be extended to user self-defined in next version.
+ * @const {!Array<!Object<{
+ *    source: string,
+ *    action: string,
+ *    target: string,
+ *    response: string
+ *  }>>}
+ * */
+genotet.linkManager.LINK_SETTINGS = [
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My Expression Matrix',
+    response: 'addProfile'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My Expression Matrix',
+    response: 'addProfile'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My Genome Browser',
+    response: 'locus'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My Genome Browser',
+    response: 'locus'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'locus'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'locus'
+  }
+];
+
+/** @const {string} */
+genotet.linkManager.DEFAULT_MAPPING_FILENAME = 'Direct Mapping';
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.linkManager.links = {};
+  genotet.data.mappingFile['gene-binding'] =
+    genotet.linkManager.DEFAULT_MAPPING_FILENAME;
+};
+
+/**
+ * Set up the link manager.
+ * @private
+ */
+genotet.linkManager.link_ = function() {
+  for (var linkViewName in genotet.linkManager.links) {
+    var linkView = genotet.viewManager.views[linkViewName];
+    var actions = genotet.linkManager.links[linkViewName];
+    $(linkView)
+      .on('genotet.nodeClick', function(event, node) {
+        actions['nodeClick'].forEach(function(obj) {
+          obj.target.signal(obj.response, node);
+        });
+      })
+      .on('genotet.edgeClick', function(event, edge) {
+        actions['edgeClick'].forEach(function(obj) {
+          obj.target.signal(obj.response, edge);
+        });
+      });
+  }
+};
+
+/**
+ * Register link action and response for all the views.
+ */
+genotet.linkManager.registerAllViews = function() {
```

Comment:
A better pattern is to call this registerLinks and pass in the LINK_SETTINGS as an array input